### PR TITLE
Honda ssc debug

### PR DESCRIPTION
**Pull request by cullenstoneAS**
add missing messages to support Honda SSC
add support for global_rpt_2 for /pacmod/enabled topic

### CODE REVIEW

**Review by kmushty-as — APPROVED** [review]

File: src/pacmod3_ros_msg_handler.cpp [hunk 1]
```diff
@@ -125,15 +125,43 @@ Pacmod3RosMsgHandler::Pacmod3RosMsgHandler(uint32_t dbc_major_version)
...
+  pub_functions[BRAKE_AUX_RPT_CANID] = std::bind(&Pacmod3RosMsgHandler::ParseAndPublishType<pacmod3_msgs::BrakeAuxRpt>, this, std::placeholders::_1, std::placeholders::_2);
+  pub_functions[HEADLIGHT_AUX_RPT_CANID] = std::bind(&Pacmod3RosMsgHandler::ParseAndPublishType<pacmod3_msgs::HeadlightAuxRpt>, this, std::placeholders::_1, std::placeholders::_2);
+  pub_functions[SHIFT_AUX_RPT_CANID] = std::bind(&Pacmod3RosMsgHandler::ParseAndPublishType<pacmod3_msgs::ShiftAuxRpt>, this, std::placeholders::_1, std::placeholders::_2);
+  pub_functions[STEERING_AUX_RPT_CANID] = std::bind(&Pacmod3RosMsgHandler::ParseAndPublishType<pacmod3_msgs::SteeringAuxRpt>, this, std::placeholders::_1, std::placeholders::_2);
+  pub_functions[TURN_AUX_RPT_CANID] = std::bind(&Pacmod3RosMsgHandler::ParseAndPublishType<pacmod3_msgs::TurnAuxRpt>, this, std::placeholders::_1, std::placeholders::_2);
+  pub_functions[WIPER_AUX_RPT_CANID] = std::bind(&Pacmod3RosMsgHandler::ParseAndPublishType<pacmod3_msgs::WiperAuxRpt>, this, std::placeholders::_1, std::placeholders::_2);
+
+  // Aux Reports
```
Comment (icolwell-as) [inline src/pacmod3_ros_msg_handler.cpp]: Component Reports

File: src/pacmod3_nodelet.cpp [hunk 2]
```diff
@@ -47,7 +47,9 @@ void Pacmod3Nl::onInit()
   all_system_statuses_pub = nh_.advertise<pacmod3_msgs::AllSystemStatuses>("all_system_statuses", 20);
 
   global_rpt_pub = nh_.advertise<pacmod3_msgs::GlobalRpt>("global_rpt", 20);
-  component_rpt_pub = nh_.advertise<pacmod3_msgs::ComponentRpt>("component_rpt", 20);
+  component_rpt_00_pub = nh_.advertise<pacmod3_msgs::ComponentRpt>("component_rpt_00", 20);
```
Comment (icolwell-as) [inline src/pacmod3_nodelet.cpp]: Can you add these new topics to the README documentation?

**Review by icolwell-as — CHANGES_REQUESTED** [review]

File: src/pacmod3_ros_msg_handler.cpp (hunk 1 above)
Comment (cullenstoneAS) [reply src/pacmod3_ros_msg_handler.cpp]: thanks

File: src/pacmod3_nodelet.cpp (hunk 2 above)
Comment (cullenstoneAS) [reply src/pacmod3_nodelet.cpp]: yep!

File: include/pacmod3/pacmod3_nodelet.h [hunk 3]
```diff
@@ -141,7 +141,9 @@ class Pacmod3Nl : public nodelet::Nodelet
   ros::Publisher all_system_statuses_pub;
   ros::Publisher global_rpt_pub;
   ros::Publisher global_rpt_2_pub;
```
Comment (jilinzhouas) [inline include/pacmod3/pacmod3_nodelet.h]: All those publisher variables stored into `pub_tx_list` with `std::move` should be deleted here.

File: src/pacmod3_nodelet.cpp [hunk 4]
```diff
@@ -62,7 +64,9 @@ void Pacmod3Nl::onInit()
   vin_rpt_pub = nh_.advertise<pacmod3_msgs::VinRpt>("vin_rpt", 5);
 
   pub_tx_list.emplace(GLOBAL_RPT_CANID, std::move(global_rpt_pub));
```
Comment (jilinzhouas) [inline src/pacmod3_nodelet.cpp]: After std::move, these independent publish variables are no longer usable. How about create them as local variables in this function.

Comment (icolwell-as) [reply src/pacmod3_nodelet.cpp]: I agree we should improve this, but maybe create a separate issue since it's unrelated to the changes in this PR.

**Review by icolwell-as — APPROVED** [review]

File: src/pacmod3_nodelet.cpp (hunk 4 above)
Comment (cullenstoneAS) [reply src/pacmod3_nodelet.cpp]: @icolwell-as sorry I didn't read your comment before I changed it, should I revert the changes for this PR?

Comment (icolwell-as) [reply src/pacmod3_nodelet.cpp]: No worries! if the work is done, all the better, just didn't want to hold the PR on something unrelated.
All good! merge at will

**Pull request merged**